### PR TITLE
Change to exit state when fatal error is raised

### PR DESCRIPTION
**Pull request by gustavosbarreto**

### CODE REVIEW

File: states.go [hunk 1]
```diff
@@ -52,6 +52,8 @@ const (
 	// UpdateHubStateWaitingForReboot is set when the agent is waiting
 	// for reboot
 	UpdateHubStateWaitingForReboot
+	// UpdateHubStateFinal is set when the daemon is about to quit
+	UpdateHubStateFinal
```
Comment (otavio) [inline states.go]: Isn't better to call it Exit? I believe it makes clear that this is going to quit the daemon (or Quit).

Comment (gustavosbarreto) [inline states.go]: Which one do you think is better?

StateFinal
StateExit
StateDone
StateQuit

Comment (otavio) [inline states.go]: I would take Quit. What do you think?

Comment (gustavosbarreto) [inline states.go]: I think StateExit is better.

**Pull request merged**